src/core/model/Stroke.cpp

```diff
@@ -259,9 +259,9 @@ void Stroke::setLastPoint(const Point& p) {
 }
 
 void Stroke::addPoint(const Point& p) {
-    this->points.emplace_back(p);
     updateBounds(Element::x, Element::y, Element::width, Element::height, Element::snappedBounds, p,
                  hasPressure() ? p.z / 2.0 : this->width / 2.0);
+    this->points.emplace_back(p);
```

Comment:
I am wrong here. As Point don't have move constructor `p` is copied when emplaced_back. The problem is that a Point can be constructed with the `z` attribute uninitialized. I will remove this commit - and ensure `Point(x,y)` initialize the z component to Point::NO_PRESSURE.

Reply:
No I am not wrong here!.. `Point` has an implicit move constructor `Point(const &&p)`. So `p` is moved after the call to `emplace_back` and therefore the `z` component of p is uninitialized - even `p` is a const reference to a Point.

Reply:
I don't quite understand what the point is. Could you dumb it down for me?

Reply:
You are correct - the copy construct are called and not the move constructor, z is initialized

Reply:
The main difference between `emplace` and `push_back` is, that `emplace` allows the type to be constructed in the container from any constructor in-place. It basically calls `new (address_in_my_buffer) Type(std::forward<Args>(args)...);` For push back, `emplace` is also called, but with the copy/move constructor. This may result in an additional move or copy, which is mostly optimized out. 

==

That a move "resets" the moved from object is a common misunderstanding. It is only guaranteed, that the value is in a valid and safely destructible state. For most cases, a move actually does nothing. And why should it, most of them are destructed anyway. For trivially copiable types, a move is the same as a copy. Generally speaking, the access of a member of an moved-from type is undefined for default and implicit move constructors, but implementation defined for explicitly defined move constructors.

==

The default initializer only initializes the member, when the called constructor does not do it in its initialization list.
On any move, there is no call of any constructor on the moved from object, therefore, no change happens to that value.
That value can only be changed over the moved-to move constructor, which does nothing when the move constructor is defaulted or implicit.